server/controllers/shoppingList.js

```diff
@@ -139,18 +140,22 @@ const getProductsForGivenList = (req, resp) => {
         { purchaserIds: req.user._id }
       ]
     },
-    'products',
     (err, documents) => {
       if (!documents) {
-        return resp.status(404).send('Products not found for given list id!');
+        return resp.status(404).send('Data not found for given list id!');
       }
 
       if (err) {
         return resp.status(404).send({ message: err.message });
       }
 
-      const { products } = documents[0];
-      resp.status(200).json(products);
+      const { _id, isArchived } = documents[0];
+
+      if (isArchived) {
+        return resp.status(200).json({ _id, isArchived });
+      }
+
+      resp.status(200).json(documents[0]);
```

Comment:
here, once you return object and once documents[0]. Can it be consistent and always return an object like in line 155?

Reply:
documents[0] is an object